### PR TITLE
docs: create I18N_USER_GUIDE.md

### PR DESCRIPTION
**Pull request by liuxiran**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [x] Improve performance

- Related issues

___
### Bugfix
- Description

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

A clearer description about i18n refer to the Proposal in https://github.com/apache/apisix-dashboard/issues/331

### CODE REVIEW

File: I18N_MANUAL.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+- please put **the global locales** under `src/locales`.
+- please put **each page locales** under `src/pages/$PAGE/locales` folder.
+- please put **self-packaged component locales** under `src/components/$COMPONENT/locales` folder. (This part cannot be loaded automatically, it needs to be additionally quoted in the global locales file)
+
+## how to name the locales key：
+
+the key can be like this : [basicModule].[moduleName].[elementName].[...desc]
+
+- basic keys (the first tow levels) divide according to the directory structure: eg: `app.pwa`, `page.consumer`, `component.actionBar`
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:34]: what's `the first tow levels`?

Comment (juzhiyuan) [reply I18N_MANUAL.md:34]: e.g `xxx`

File: I18N_MANUAL.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+- sub keys divide according to element +description eg: `app.pwa.message.offline`, `component.actionBar.button.nextStep`
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:36]: The subkeys are divided into `$element + $description` style, e.g `xxx`

File: I18N_MANUAL.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+* If the the text is the part of a element, we can use [elementNameProps] eg: `page.consumer.proTableColumns.username`
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:38]: Please do a global search about `e.g`

File: I18N_MANUAL.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+* If there are two or more same level part locales of a element, we can add number suffix eg: `page.route.form.itemRulesExtraMessage1.path`, `page.route.form.itemRulesExtraMessage2.path`
+
+- common texts, we should not repeat in other part, and the common locale key omit [elementName] would be better. If the text is used in two or more places inside the module, it`s recommended that this text be shared in the module eg:`page.route.parameterPosition`If the text is used in two or more places between modules, it is recommended that the text be shared globally, and add`global`as the moduleName eg:`component.global.confirm`
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:41]: Please take a look at this paragraph:

image

File: I18N_MANUAL.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+## Global locale keys
+
+we have already define many global keys, before you do i18n, you can refer to [them](https://github.com/apache/apisix-dashboard/blob/master/src/locales/zh-CN/component.ts).
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:45]: We have already defined.

Comment (juzhiyuan) [reply I18N_MANUAL.md:45]: to those

File: I18N_MANUAL.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+- **Form**
+
+| element   | props          | locale subKey                 |
+| --------- | -------------- | ----------------------------- |
+| Form.Item | label          | form.itemLabel                |
+| Form.Item | rules.required | form.itemRulesRequiredMessage |
+| Form.Item | rules.pattern  | form.itemRulesPatternMessage  |
+| Form.Item | extra          | form.itemExtraMessage         |
+
+**eg:**
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:58]: Example

File: I18N_MANUAL.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+```javascript
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:60]: js

File: I18N_MANUAL.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+'page.route.form.itemRulesPatternMessage.apiNameRule': '最大长度100，仅支持字母、数字、- 和 _，且只能以字母开头',
+```
+
+- **Input**
+
+| element | props       | locale subKey     |
+| ------- | ----------- | ----------------- |
+| Input   | placeholder | input.placeholder |
+
+**eg:**
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:72]: Example

File: I18N_MANUAL.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+```javascript
```
Comment (juzhiyuan) [inline I18N_MANUAL.md:74]: js

File: I18N_MANUAL.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+
+# Apache APISIX Dashboard I18N Manual
```
[the remark this answers is not in the document]
Comment (liuxiran) [reply I18N_MANUAL.md:20]: `User Guide` is better，I will change the file name to `I18N_USER_GUIDE.md` at the same time.

**Comment by liuxiran** [comment]
> LGTM, but we should fix those typos before merging :DD

thanks for your suggestions🤝
which not only make this document more professional, but also  teach me some skills in writing English documents.

I `m  fixing typos refer to this doc, and it will take some time to finish it, later I'll submit pr.

**Comment by juzhiyuan** [comment]
I see you just did a force-push, and I cannot see what has been changed, so this PR could be merged, right?

File: I18N_USER_GUIDE.md [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+the key can be like this : [basicModule].[moduleName].[elementName].[...desc]
+
+- what's the first tow levels? e.g: `app.pwa`, `page.consumer`, `component.actionBar`
+
+- The subkeys are divided into $element + $description style e.g: `app.pwa.message.offline`, `component.actionBar.button.nextStep`
+
+  - If the the text is the part of a element, we can use [elementNameProps] e.g: `page.consumer.proTableColumns.username`.
+  - If there are two or more same level part locales of a element, we can add number suffix e.g: `page.route.form.itemRulesExtraMessage1.path`, `page.route.form.itemRulesExtraMessage2.path`.
+
+- common texts, we should not repeat in other part, and the common locale key omit [elementName] would be better.
```
Comment (liuxiran) [inline I18N_USER_GUIDE.md:41]: sorry to do a force-push, I will pay attention next time。
This paragraph needs check, and others fixed refer to your suggestions。
@juzhiyuan

File: I18N_USER_GUIDE.md [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+
+  - If the text is used in two or more places inside the module, we would recommend to share the text in the module, e.g:`page.route.parameterPosition`.
+  - If the text is used in two or more places between modules, we would recommend to share the text globally, and add`global`as the moduleName,git e.g:`component.global.confirm`.
+
```
Comment (liuxiran) [inline I18N_USER_GUIDE.md:45]: sorry to do a force-push, I will pay attention next time。
This paragraph needs check（from line 40 to44）, and others fixed refer to your suggestions。
@juzhiyuan

Comment (juzhiyuan) [reply I18N_USER_GUIDE.md:45]: If the text is used in two or more places between modules, we would recommend sharing the text globally, and add `global` as the moduleName, e.g `component.global.confirm`.

Comment (liuxiran) [reply I18N_USER_GUIDE.md:45]: > If the text is used in two or more places between modules, we would recommend sharing the text globally, and add `global` as the moduleName, e.g `component.global.confirm`.

ok, thanks, I modified.

**Pull request merged**